### ISSUE TITLE
Anchor links within a page do not function correctly

### ISSUE THREAD

**Issue by TangentFoxy** [alangrainger/share-note#76]
In a note named `Note` with a heading like `## Subsection`, a link formatted as `[[Note#Subsection]]` creates a link only to the note itself instead of the named section.

While a link like `[[#Subsection]]` navigates to the correct location within a note, it does not update the address bar of the browser to the correct anchor.

**Comment by FeralFlora** [comment]
I would love if this were to work as well. It would make table of contents much more useful in long articles (two-way navigation). And it would be very convenient to be able to link to specific sections of an article.

I use the Automatic Table of Contents plugin to create tables of contents. These links work, but the headings themselves are **missing the IDs necessary for navigating back** to the ToC from the headings link anchor (since there is none).

**Comment by alangrainger** [comment]
This could be done, the problem is that anchor links normally work using the hash part of the URL. In the case of this plugin, we're using that for the decrypt key.

What could be done is use a query param like `&anchor=abc` instead, and use Javascript to jump to the right place. Biggest issue is that this would cause the page to reload on each click.

Either that or have clickable heading links only available for unencrypted pages.

**Comment by TangentFoxy** [comment]
I see a release says this was fixed, but I just tested it and it isn't working. Was that fix to only support anchor tags *on the same page*? What I just tested was using them across different notes, not within the same note.

So I just tested again, and I see it works for links in the format `[[#this is a heading]]` but not `[[Note Name#this is a heading]]` or like I expected from other notes. That said, if it's infeasable to do more, this is still really good, thank you!

**Comment by alangrainger** [comment]
> Was that fix to only support anchor tags on the same page

Yes, that's correct. This issue was for "Anchor links *within* a page". I will look at inter-page anchor links too.

**Comment by alangrainger** [comment]
> Was that fix to only support anchor tags on the same page?

Yes,  that's correct. I can look at inter-page anchor links too.

**Comment by arteny** [comment]
links like `[[#Heading1|Heading 1]]` (with description) not works for me
